index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/SearchNviCandidatesHandlerTest.java

```diff
@@ -460,6 +462,23 @@ void shouldExcludeContributorsInSearchResponse() throws IOException {
         .search(argThat(argument -> argument.excludeFields().equals(expectedExcludeFields)));
   }
 
+  @Test
+  void shouldSearchByReportingPeriodWhenYearParamIsSet() throws IOException {
```

Comment:
Seems to me like this test the `id` of the search result, not that we are actually searching for reporting year, or?

Reply:
The `id` field in the response is the query string that produces this response (which should be what we actually searched for). That's all we can really test in this class, because the actual OpenSearch client is mocked and integration tests done separately.